### ISSUE TITLE
Long quick info has no scrollbar

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/vscode#40674]
- VSCode Version: Code - Insiders 1.20.0-insider (3aed233178b31fa75938aae37883fa18033f4b8b, 2017-12-20T06:12:56.284Z)
- OS Version: Linux x64 4.4.0-21-generic
- Extensions: none
---

Steps to Reproduce:

1. Paste this text:
```ts
declare function update<T>(a: T, b: Partial<T>): void;
update({
    a: 0,
    b: 0,
    c: 0,
    d: 0,
    e: 0,
}, {
    b: 3,
});
```
2. Hover over `update`.
3. Expected to get a scrollbar. Instead the text just gets cut off.

**Comment by weinand** [comment]
Yes this is a regression in Insiders:

![2017-12-21_23-49-22](https://user-images.githubusercontent.com/1898161/34278203-b92fb2ea-e6a9-11e7-8e14-d05f8078aa01.gif)

Here the behaviour in Stable:

![2017-12-21_23-43-49](https://user-images.githubusercontent.com/1898161/34278221-cac96406-e6a9-11e7-91d8-da44d927954a.gif)

**Comment by joaomoreno** [comment]
Broken by https://github.com/Microsoft/vscode/commit/bfe83a84d1122b91315301d8b38a19aaf04a9523 🤔

**Comment by joaomoreno** [comment]
Turns out there is a different between `TPromise.join` and `Promise.all`, very similar to the difference between promise ctors:

- In the `TPromise` version, the [`join` `then` callback](https://github.com/Microsoft/vscode/blob/c847df3d259f64cd763d8a8d24ecbcb4d697a968/src/vs/base/browser/htmlContentRenderer.ts#L129) gets called as soon as [`signalInnerHTML` fires](https://github.com/Microsoft/vscode/blob/c847df3d259f64cd763d8a8d24ecbcb4d697a968/src/vs/base/browser/htmlContentRenderer.ts#L166). When `renderMarkdown` returns, all colorized code is in the hover DOM. The hover widget tells the scrollbar to scan the DOM to compute the content height and everything works fine.
- In the `Promise` version, the [`join` `then` callback](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/htmlContentRenderer.ts#L129) only gets called on the next execution frame... so the hover widget tells the scrollbar to scan the DOM before any of the colorized code is placed in the DOM.

As a fix for this particular issue, I've added a `codeBlockRenderCallback` option to the markdown render function, which gets called every time colorised code gets inserted into the DOM. Ugly, but effective. @mjbvz
